src/server/models/db/migrations/20230214150000_populate_events_for_published_list_items/migration.sql

```diff
@@ -0,0 +1,12 @@
+-- populate Event records for list items in PUBLISHED status that were published before the Event table was introduced
+
+INSERT INTO "Event" ("listItemId", "time", "type", "jsonData")
+select "ListItem"."id"        as "listItemId",
+       "ListItem"."updatedAt" as "time",
+       'PUBLISHED'            as "type",
+       concat('{"eventName": "publish"}') ::json as "jsonData"
```

Comment:
there's whitespace here `concat('{"eventName": "publish"}') ::json`. I'm not sure what you are concat-ing with.

I had a look at the audit table, it dates back to 2021. I think we'll be able to use this instead. 

```sql
-- create a CTE (temp table) joining Audit to Events. Returns a row when Audit is missing a matching (publish) event. 
-- Only select the relevant columns and rename them to match `Event` insert.
with AuditWithoutMatchingListItemEvent as (select (a."jsonData" ->> 'itemId')::integer as "listItemId",
                                                  a."createdAt"                        as time,
                                                  a."jsonData"
                                           from "Audit" a
                                                    left outer join "Event" e on (a."jsonData" ->> 'itemId')::integer = e."listItemId"
                                           where e.id is null
                                             and a."jsonData" ->> 'eventName' = 'publish'),

-- Create a CTE (temp table) joining the above audits to a ListItem. Returns when a row has a ListItem. (i.e. ListItem hasn't been deleted)
-- Only select the relevant columns (from above).
     AuditsToMigrate as (select a.*
                         from AuditWithoutMatchingListItemEvent a
                                  left outer join "ListItem" li on a."listItemId" = li.id
                         where li.id is not null)


insert
into "Event" ("listItemId", "time", "type", "jsonData")
select "listItemId", time, 'PUBLISHED', "jsonData"
from AuditsToMigrate;
rollback;

```

Reply:
as discussed, the migration script has now been converted to use the data derived from Audit